### PR TITLE
Field CRUD

### PR DESCRIPTION
**Pull request by KevinBigelow**
The Bite Sized PRS for reference 🍪
It would make sense to look through them in order

1. Nesting base field modules: https://github.com/neutroninteractive/neutron-web/pull/24
2. Text field crud: https://github.com/KevinBigelow/neutron-web/pull/4
3. Option field crud: https://github.com/KevinBigelow/neutron-web/pull/5
4. Options crud: https://github.com/KevinBigelow/neutron-web/pull/6
5. Additional base field refinements: https://github.com/KevinBigelow/neutron-web/pull/7

### CODE REVIEW

**Comment by KevinBigelow** [comment]
We need to merge https://github.com/neutroninteractive/neutron-web/pull/33 first

File: src/components/forms/base-fields/create.vue [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<template>
+  <modal-template :show="showModal" @close="close">
+    <template v-slot:header>Create Field</template>
+    <template v-slot:body>
+      <validation-observer v-slot="{ handleSubmit }" ref="form">
+        <form @submit.prevent="handleSubmit(submitForm)">
+          <v-text-field v-model="name" rules="required" field_id="createBaseTextFieldName" field_label="Name" class="field-group"></v-text-field>
+          <v-text-field v-model="label" rules="required" field_id="createBaseTextFieldLabel" field_label="Label" class="field-group"></v-text-field>
+          <text-field v-model="desc" field_id="createBaseTextFieldDesc" field_label="Description" class="field-group"></text-field>
```
Comment (emilyfjonaitis) [inline src/components/forms/base-fields/create.vue:9]: nonono abbreviations kevin

File: src/components/forms/base-fields/create.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+<script>
+import { mapActions, mapGetters, mapMutations } from 'vuex'
+import { enterKeyListener } from '@/mixins/enterKeyListener'
+
+export default {
+  data () {
+    return {
+      name: '',
+      label: '',
+      desc: '',
```
Comment (emilyfjonaitis) [inline src/components/forms/base-fields/create.vue:30]: change to description

File: src/components/forms/base-fields/text-fields/update.vue [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+<script>
+import { mapActions, mapGetters, mapMutations } from 'vuex'
+
+export default {
+  data () {
+    return {
+      name: '',
+      label: '',
+      desc: '',
```
Comment (emilyfjonaitis) [inline src/components/forms/base-fields/text-fields/update.vue:26]: changes to description

Comment (KevinBigelow) [reply src/components/forms/base-fields/text-fields/update.vue]: this is updated @emilyfjonaitis

File: src/components/forms/base-fields/create.vue (hunk 2 above)
Comment (KevinBigelow) [reply src/components/forms/base-fields/create.vue]: this is updated @emilyfjonaitis

File: src/components/forms/base-fields/create.vue (hunk 1 above)
Comment (KevinBigelow) [reply src/components/forms/base-fields/create.vue]: this is updated @emilyfjonaitis - hit the resolve button if you agree

File: src/store/modules/buyers.js [hunk 4]
```diff
@@ -35,6 +35,7 @@ const actions = {
     await axios.get(`/buyers/${id}/`)
       .then(response => {
         commit('SET_CURRENT_BUYER', response.data)
+        commit('SET_CURRENT_FORM', response.data.form) // Not convinced that this is correct. Mutation could be called on field/list component
```
Comment (mickmasterflex) [inline src/store/modules/buyers.js:38]: this still needs to be looked at? or is it good to go?

**Pull request merged**